Src/LinearSolvers/MLMG/AMReX_MLEBABecLap_2D_K.H

```diff
@@ -41,32 +45,56 @@ void mlebabeclap_adotx (Box const& box, Array4<Real> const& y,
             Real apym = apy(i,j,k);
             Real apyp = apy(i,j+1,k);
 
-            Real fxm = bX(i,j,k,n)*(x(i,j,k,n)-x(i-1,j,k,n));
+            Real fxm = bX(i,j,k,n) * (x(i,j,k,n)-x(i-1,j,k,n));
             if (apxm != 0.0 and apxm != 1.0) {
                 int jj = j + static_cast<int>(amrex::Math::copysign(1.0,fcx(i,j,k)));
                 Real fracy = (ccm(i-1,jj,k) || ccm(i,jj,k)) ? amrex::Math::abs(fcx(i,j,k)) : 0.0;
-                fxm = (1.0-fracy)*fxm + fracy*bX(i,jj,k,n)*(x(i,jj,k,n)-x(i-1,jj,k,n));
+                if (beta_on_center and phi_on_center) 
+                   fxm = (1.0-fracy)*fxm + fracy*bX(i,jj,k,n)*(x(i,jj,k,n)-x(i-1,jj,k,n));
+                else if (beta_on_centroid and phi_on_center) 
+                   fxm = bX(i,j,k,n) * ( (1.0-fracy)*(x(i, j,k,n)-x(i-1, j,k,n))
+                                           + fracy  *(x(i,jj,k,n)-x(i-1,jj,k,n)) );
+                else if (beta_on_centroid and phi_on_centroid) 
+                   amrex::Abort("beta_on_centroid and phi_on_centroid is still a WIP");
```

Comment:
We might want to move these type of Abort calls out of the kernel functions for performance reason.

Reply:
I have moved the tests on "phi_on_centroid" out of the individual loops.